Add info about unsyncable docs (#1188)
mongodb/docs-realm: source/mongodb.txt

      { $unset: { _id_realm_transaction: "" } }
   )

Realm Sync Schema Data Consistency
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> If the collection in which you create and modify objects is also accessed by Realm 
> :ref:`Sync clients <sync>`, you must ensure that the data conforms to the 

Reviewer: ```suggestion
If the collection in which you create and modify objects is also accessed by
:ref:`{+sync+} clients <sync>`, you must ensure that the data conforms to the 
```

> Realm :ref:`object schema <object-schema>` defined on the collection. Consider 

Reviewer: ```suggestion
{+service-short+} :ref:`object schema <object-schema>` defined on the collection. Consider 
```

the following scenario:

> - You have a collection of objects that is used by {+realm+} Sync clients. The 

Reviewer: ```suggestion
- You have a collection of objects that is used by {+sync+} clients. The 
```
realm replacement is for lowercase-r realm database instances

  **object schema** for the collection looks like this:

  .. code-block:: js
     :copyable: false

     type Person = {
       _id: BSON.ObjectId;
       name: string;
       address: string;
       age: number;
     };
  

> - Using a Realm client (say, {+compass+}), you create and insert a document 

Reviewer: ```suggestion
- Using a {+service-short+} client (say, {+compass+}), you create and insert a document 
```

  into the collection. The ``age`` property of this document has a value of 
  "forty two". 
  
- Because of MongoDB's flexible schema, there is no error when inserting the 
  document. However, because the the Sync object schema defines the ``age`` 
  property as type ``number``, the object fails schema validation and is not 
  pushed to Sync.

- The collection is now divergent: Sync clients do not see the new document, but 
  {+service+} clients do. 

The same scenario holds true if a valid document is modified outside of Sync 
such that one or more properties no longer conform to the Sync schema. 

Any document that fails the object schema validation is added to the 

> ``__realm_sync.unsynced_documents`` collection (along with the reason for the 

Reviewer 2: Can we add a note or something to make sure people realize it is ok if they read from this collection / db but they should never write to it?

failure) and will also be logged in the 

> Realm UI ``Logs`` console. If there are 10,000 or more documents that can't be 

Reviewer 2: Turns out this number is actually 100,000 :)

Author: eh, what's an extra digit amongst friends?

synced, the Realm app is paused.

To fix this problem, you must **replace** the invalid document with one that 
conforms to the object model. Updating the existing document will not add it to 
Sync. 

.. note:: Summary

   If your data is used by Sync clients but can also be created or 

>    modified outside of Realm Sync, you must ensure those creations and modifications 

Reviewer: ```suggestion
   modified outside of {+sync+}, you must ensure those creations and modifications 
```
